From fd8ca1c4336b29bbe4e829a01c13707a5c4035f7 Mon Sep 17 00:00:00 2001
From: Ravi Kumar <ravi1.kumar@amd.com>
Date: Thu, 23 Aug 2018 11:57:15 +0530
Subject: [PATCH 4114/4131] Fix for ROCm not working on dGPU issue

Removed the BUILD_AS_DKMS macro around the pci_enable_atomic_ops_to_root
API which is necessary to get ROCm work on dGPUs.

Signed-off-by: Ravi Kumar <ravi1.kumar@amd.com>
---
 drivers/gpu/drm/amd/amdkfd/kfd_device.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdkfd/kfd_device.c b/drivers/gpu/drm/amd/amdkfd/kfd_device.c
index 14275b1..a9ad2a8 100755
--- a/drivers/gpu/drm/amd/amdkfd/kfd_device.c
+++ b/drivers/gpu/drm/amd/amdkfd/kfd_device.c
@@ -366,14 +366,12 @@ struct kfd_dev *kgd2kfd_probe(struct kgd_dev *kgd,
 		 * 32 and 64-bit requests are possible and must be
 		 * supported.
 		 */
-#if defined(BUILD_AS_DKMS)
                 if (pci_enable_atomic_ops_to_root(pdev) < 0) {
                         dev_info(kfd_device,
                                 "skipped device %x:%x, PCI rejects atomics",
                                  pdev->vendor, pdev->device);
                         return NULL;
                 }
-#endif
 	}
 
 	kfd = kzalloc(sizeof(*kfd), GFP_KERNEL);
-- 
2.7.4

